### ISSUE TITLE
[Bug] Missing keys are returned without error as`Bytes`

### ISSUE THREAD

**Issue by stepantubanov** [aembke/fred.rs#164]
Redis version - 6.2.11
Platform - mac
Using Docker and/or Kubernetes - yes
Deployment type - cluster

**Describe the bug**

```rust
let conn = self.pool.next();
let serialized: Bytes = conn.get("does-not-exist").await?;
```

This code actually runs without error (redis key doesn't exist) which seems wrong. It yields this value: `b"$-1\r\n"`. Shouldn't it return an error?

**Comment by aembke** [comment]
This should be fixed in 7.x (starting at https://crates.io/crates/fred/7.0.0-rc.1). There's also a new FF `default-nil-types` that might be helpful here. I'll leave this open until 7.0.0 is released though.

**Comment by aembke** [comment]
This should be fixed in [7.0.0](https://crates.io/crates/fred/7.0.0). Please let me know if you still see any issues here though.
